uniresis/src/uniresis.cpp

```diff
@@ -89,14 +89,20 @@ class uniresis_t::updater_t : public std::enable_shared_from_this<uniresis_t::up
         subscope(),
         executor(),
         timer(executor.asio()),
-        log(std::move(log))
+        log(log)
     {}
 
+    ~updater_t() {
+        scope->close();
+        subscope->close();
+        timer.cancel();
+    }
+
     auto
     notify() -> void {
         COCAINE_LOG_DEBUG(log, "schedule resource notification on `{}` ...", path);
         scope = unicorn->create(
-            std::bind(&updater_t::on_create, shared_from_this(), ph::_1),
+            std::bind(&updater_t::on_create, this, ph::_1),
```

Comment:
No, you can't do this unless you own the I/O loop.

Reply:
I own cancel token and it is thread safe - after close call no more callbacks will be dispatched and executing ones will end before close return.

Reply:
Imagine a situation when this callback is called while we're destructing ourself.

Reply:
so what? destruction of members do not start until dtor body has been executed, and dtor cancelss callback.